WalletWasabi/WabiSabi/Client/BobClient.cs

```diff
@@ -0,0 +1,31 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	public class BobClient
+	{
+		public BobClient(Guid roundId, ArenaClient arenaClient)
+		{
+			RoundId = roundId;
+			ArenaClient = arenaClient;
+		}
+
+		public Guid RoundId { get; }
+		public ArenaClient ArenaClient { get; }
```

Comment:
`ArenaClient` has to be private as in `AliceClient`. Anyway, clients have to create their own `ArenaClient` instances because the `IRequestHandler` is different for Alice, Bod and Satoshi. What the all share is the `CredentialPool`.

Reply:
It seems to be impossible to share the `CredentialPool` with the current implementation. CredentialPool should be a ctor parameter in ArenaClient?

Reply:
yes, and I think there's a good chance it will make sense to have isolated credential pools for them created and managed by the overlying client abstraction when the credential dependencies are done, otherwise they would have to know which credentials to use to ensure they don't interfere with each other (if a client uses a credential needed by another client might cause it to get stuck unable to perform its credential request with <= k credentials)

Reply:
Yes.

Reply:
https://github.com/zkSNACKs/WalletWasabi/pull/5581

Reply:
Merged